File: 2130-maximum-twin-sum-of-a-linked-list/2130-maximum-twin-sum-of-a-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def pairSum(self, head: Optional[ListNode]) -> int:
        node, stack, big = head, [], 0
  
        while node:
            stack.append(node.val)
            node = node.next
            
        n = len(stack) - 1
        for i in range(n):
            big = max(big,stack[i]+stack[n-i])
        return big

File: 2130-maximum-twin-sum-of-a-linked-list/2130-maximum-twin-sum-of-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int pairSum(ListNode* head) {
        vector<int> s;
        while(head) {
            s.push_back(head->val);
            head = head->next;
        }
        int n = s.size(), ans = 0;
        for (int i = 0; i < (n/2); i++) {
            ans = max(ans, s[i] + s[n-i-1]);
        }
        return ans;
    }
};